src/RolloutScorer/RolloutScorer/Providers/RolloutScorerProvider.cs

```diff
@@ -11,103 +11,111 @@
+            _buildHttpClient = buildHttpClient;
+        }
 
-            foreach (string buildDefinitionId in RepoConfig.BuildDefinitionIds)
+        public async Task InitAsync(Models.RolloutScorer rolloutScorer)
+        {
+            foreach (string buildDefinitionId in rolloutScorer.RepoConfig.BuildDefinitionIds)
             {
-                string azdoQuery = $"https://dev.azure.com/{RepoConfig.AzdoInstance}/" +
-                    $"{AzdoConfig.Project}/_apis/build/builds?definitions={buildDefinitionId}&branchName={Branch}" +
-                    $"&minTime={rolloutStartTimeUriString}&maxTime={rolloutEndTimeUriString}&api-version=5.1";
-                Utilities.WriteDebug($"Querying AzDO API: {azdoQuery}", Log, LogLevel);
-                JObject responseContent = await GetAzdoApiResponseAsync(azdoQuery);
+                // organization = rolloutScorer.RepoConfig.AzdoInstance
+                // project = rolloutScorer.AzdoConfig.Project
+                // definitions = buildDefinitionId
+                // branchName = rolloutScorer.Branch
+                // minTime = 
+                // maxTime = 
+
+                var azDOBuilds = await _buildHttpClient.GetBuildsAsync(
+                    definitions: new []{ Convert.ToInt32(buildDefinitionId) }, 
+                    minFinishTime: rolloutScorer.RolloutStartDate.UtcDateTime, 
+                    maxFinishTime: rolloutScorer.RolloutEndDate.UtcDateTime, 
+                    branchName: rolloutScorer.Branch);
+                _logger.LogInformation($"Querying AzDO API");
 
                 // No builds is a valid case (e.g. a failed rollout) and so the rest of the code can handle this
                 // It still is potentially unexpected, so we're going to warn the user here
-                if (responseContent.Value<int>("count") == 0)
+                if(azDOBuilds.Count == 0)
                 {
-                    Utilities.WriteWarning($"No builds were found for repo '{RepoConfig.Repo}' " +
-                            $"(Build ID: '{buildDefinitionId}') during the specified dates ({RolloutStartDate} to {RolloutEndDate})", Log);
+                    _logger.LogWarning($"No builds were found for repo '{rolloutScorer.RepoConfig.Repo}' " +
+                            $"(Build ID: '{buildDefinitionId}') during the specified dates ({rolloutScorer.RolloutStartDate} to {rolloutScorer.RolloutEndDate})");
                 }
 
-                JArray builds = responseContent.Value<JArray>("value");
-                foreach (JToken build in builds)
+                foreach (Microsoft.TeamFoundation.Build.WebApi.Build build in azDOBuilds)
                 {
-                    BuildBreakdowns.Add(new ScorecardBuildBreakdown(build.ToObject<BuildSummary>()));
+                    // TODO: Convert Build into BuildSummary
+                    rolloutScorer.BuildBreakdowns.Add(new Models.ScorecardBuildBreakdown(build.ToObject<Models.BuildSummary>()));
```

Comment:
ScorecardBuildBreakdown needs to be refactored to accept the Build type instead of BuildSummary.  Determine if BuildSummary is still necessary.